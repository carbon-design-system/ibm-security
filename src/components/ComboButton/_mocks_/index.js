/**
 * @file Combo button mocks.
 * @copyright IBM Security 2019
 */

import { ArrowRight20, FolderDetails20 } from '@carbon/icons-react';

const actions = [
  {
    label: 'Start a task',
    iconDescription: 'arrow right',
    renderIcon: ArrowRight20,
    onClick: () => {},
  },
  {
    label: 'Add to folder',
    renderIcon: FolderDetails20,
    onClick: () => {},
  },
];

export default actions;
