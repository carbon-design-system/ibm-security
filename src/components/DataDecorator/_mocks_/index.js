/**
 * @file Data decorator mocks.
 * @copyright IBM Security 2019
 */

const props = {
  type: 'IP',
  value: '10.0.0.0',
  score: 0,
  href: '#',
};
const midLine = {
  type: 'MD5',
  value: '0f3deda483df5e5f8043ea20297d243b',
  score: 30,
  href: '#',
};

export default props;

export { props, midLine };
