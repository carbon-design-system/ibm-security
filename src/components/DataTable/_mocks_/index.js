/* eslint-disable react/prop-types */
/**
 * @file Data table mocks.
 * @copyright IBM Security 2019
 */

import React from 'react';
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableHeader,
  TableRow,
  TableSelectAll,
  TableSelectRow,
} from '../../..';

const headers = [
  {
    key: 'lastSighted',
    header: 'Last sighted',
  },
  {
    key: 'name',
    header: 'Name',
  },
  {
    key: 'type',
    header: 'Type',
  },
  {
    key: 'attackPhase',
    header: 'Attack phase',
  },
  {
    key: 'method',
    header: 'Method',
  },
  {
    key: 'missing',
    header: 'Missing',
  },
];

const isSelectable = false;
const isSortable = false;

const rows = [
  {
    id: '7c34f4507156b35',
    lastSighted: '11/28/2018',
    name: 'IBM X-Force',
    type: 'IPv6',
    attackPhase: 'Initial Compromise',
    method: 'Phishing',
  },
  {
    id: '38347816a51b0c5a8',
    lastSighted: '06/28/2017',
    name: 'AppX',
    type: 'Hash',
    attackPhase: 'Establish Foothold',
    method: 'Deploy backdoor',
  },
  {
    id: '1d6116a2d1d03b4',
    lastSighted: '05/30/2018',
    name: 'Bigfix',
    type: 'URL',
    attackPhase: 'Escalate Privileges',
    method: 'Credential dumping',
  },
  {
    id: 'c52b0419981221765',
    lastSighted: '12/12/2018',
    name: 'Am I Affected',
    type: 'Hash',
    attackPhase: 'Cross-Site Scripting',
    method: 'Malicious Code',
  },
  {
    id: '697626e6f703131d',
    lastSighted: '11/11/2018',
    name: 'QRadar',
    type: 'Hash',
    attackPhase: 'SQL Injection Attack',
    method: 'SQL Vulnerabilities',
  },
  {
    id: 'd568f12fa7b7cac1',
    lastSighted: '12/09/2018',
    name: 'iris',
    type: 'Hash',
    attackPhase: 'Volume Based Attacks',
    method: 'ICMP floods',
  },
  {
    id: '697626e6f70313554',
    lastSighted: '12/10/2018',
    name: 'TISP',
    type: 'Hash',
    attackPhase: 'Application Layer Attacks',
    method: 'GET/POST floods',
  },
  {
    id: '1dd568f12fa7b7cac2',
    lastSighted: '12/11/2018',
    name: 'CLX',
    type: 'Hash',
    attackPhase: 'Protocol Attacks',
    method: 'SYN Floods',
  },
  {
    id: '697626e6f703131',
    lastSighted: '10/12/2018',
    name: 'X-Force Plus',
    type: 'Hash',
    attackPhase: 'Malicious Links',
    method: 'Social Engineering',
  },
  {
    id: 'dd568f12fa7b7cac0',
    lastSighted: '12/16/2018',
    name: 'BigFix & iris',
    type: 'Hash',
    attackPhase: 'Exploit Kits',
    method: 'Scripts',
  },
  {
    id: '697626e6f70313',
    lastSighted: '01/12/2018',
    name: 'Cases',
    type: 'Hash',
    attackPhase: 'Malicious Email',
    method: 'Social Engineering',
  },
];

const page = 1;
const pageSize = 5;
const pageSizes = [5, 10, 20];

/**
 * Render method.
 * @returns {string} The string to render.
 */
const render = ({
  rows,
  headers,
  getHeaderProps,
  getRowProps,
  getTableProps,
  getSelectionProps,
}) => (
  <TableContainer>
    <Table {...getTableProps()}>
      {headers && (
        <TableHead>
          <TableRow>
            <TableSelectAll {...getSelectionProps()} />
            {headers.map(header => (
              <TableHeader
                key={header.id}
                {...getHeaderProps({ header, isSortable })}
              >
                {header.header}
              </TableHeader>
            ))}
          </TableRow>
        </TableHead>
      )}
      <TableBody>
        {rows.map(row => (
          <TableRow {...getRowProps({ row })} key={row.id}>
            <TableSelectRow {...getSelectionProps({ row })} />
            {row.cells.map(cell => (
              <TableCell key={cell.id}>
                {cell.value ? cell.value : 'â€“'}
              </TableCell>
            ))}
          </TableRow>
        ))}
      </TableBody>
    </Table>
  </TableContainer>
);

export {
  headers,
  isSelectable,
  isSortable,
  page,
  pageSize,
  pageSizes,
  render,
  rows,
};
