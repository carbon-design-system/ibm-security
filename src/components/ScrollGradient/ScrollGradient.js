/**
 * @file Scroll gradient.
 * @copyright IBM Security 2019 - 2020
 */

import classnames from 'classnames';
import PropTypes from 'prop-types';
import React, { Component } from 'react';

import { throttle } from 'throttle-debounce';

import { getComponentNamespace } from '../../globals/namespace';
import { isClient } from '../../globals/utils/capabilities';

export const namespace = getComponentNamespace('scroll-gradient');

const scrollDirection = { X: 'X', Y: 'Y' };

// Checks if the native version is available and switch between this and the polyfill to improve performance on browsers with native support - https://www.npmjs.com/package/resize-observer-polyfill
const ResizeObserver =
  (isClient() && window.ResizeObserver) || require('resize-observer-polyfill');

class ScrollGradient extends Component {
  static propTypes = {
    /** @type {string} Fade out color. Any valid CSS color value works */
    color: PropTypes.string.isRequired,

    /** @type {string} Scroll area children */
    children: PropTypes.oneOfType([
      PropTypes.node,
      PropTypes.arrayOf(PropTypes.node),
    ]),

    /** @type {string} Optional classname */
    className: PropTypes.string,

    /** @type {string} Scroll direction */
    direction: PropTypes.oneOf(['X', 'Y']),

    /** @type {boolean} Set to true if you want to hide gradient on the start side (top or left) of scrollable element. */
    hideStartGradient: PropTypes.bool,

    /** @type {Function} Optional scroll handler */
    onScroll: PropTypes.func,

    /** @type {string} Optional classname for scroll element. */
    scrollElementClassName: PropTypes.string,

    /** @type {(element: HTMLElement) => {}} Optional function to get reference to scrollable DOM element */
    getScrollElementRef: PropTypes.func,
  };

  static defaultProps = {
    className: undefined,
    children: undefined,
    scrollElementClassName: undefined,
    direction: scrollDirection.Y,
    hideStartGradient: false,
    onScroll: () => {},
    getScrollElementRef: () => {},
  };

  /**
   * Gets the scroll state position of a given element.
   * @param {HTMLElement} element The element to get scroll state of.
   * @param {string} scrollDirection The scroll direction to get the state of.
   * @returns {string} State of scroll position.
   * @static
   */
  static getScrollState = (element, scrollDirection) => {
    switch (scrollDirection) {
      case ScrollGradient.ScrollDirection.X: {
        if (element.scrollWidth === element.clientWidth)
          return ScrollGradient.ScrollStates.NONE;
        if (element.scrollLeft === 0)
          return ScrollGradient.ScrollStates.INITIAL;
        if (element.scrollLeft + element.clientWidth === element.scrollWidth)
          return ScrollGradient.ScrollStates.END;
        return ScrollGradient.ScrollStates.STARTED;
      }

      case ScrollGradient.ScrollDirection.Y:
      default: {
        if (element.scrollHeight === element.clientHeight)
          return ScrollGradient.ScrollStates.NONE;
        if (element.scrollTop === 0) return ScrollGradient.ScrollStates.INITIAL;
        if (element.scrollTop + element.clientHeight === element.scrollHeight)
          return ScrollGradient.ScrollStates.END;
        return ScrollGradient.ScrollStates.STARTED;
      }
    }
  };

  state = {
    position: ScrollGradient.ScrollStates.NONE,
  };

  componentDidMount() {
    if (this.scrollContainer) {
      this.updateScrollState();
    }

    if (isClient()) {
      // https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver
      this.resizeObserver = new ResizeObserver(this.updateHandler);

      this.resizeObserver.observe(this.scrollContainer);
    }
  }

  componentDidUpdate() {
    if (
      this.state.position !==
      ScrollGradient.getScrollState(this.scrollContainer, this.props.direction)
    ) {
      this.updateScrollState();
    }
  }

  componentWillUnmount() {
    this.resizeObserver.disconnect();
  }

  /**
   * Saves references of scroll DOM element.
   * @param {HTMLElement} element Scroll DOM element.
   */
  setRefs = element => {
    this.scrollContainer = element;
    this.props.getScrollElementRef(element);
  };

  /** @enum Possible scroll directions  */
  static ScrollDirection = scrollDirection;

  /** @enum Scroll position states */
  static ScrollStates = {
    // No scrolling required because content fits within container.
    NONE: 'NONE',

    // Scroll position is a the start of the scrollable content.
    INITIAL: 'INITIAL',

    // Scroll position is neither at start or end of scrollable content.
    STARTED: 'STARTED',

    // Scroll position is a the end of the scrollable content.
    END: 'END',
  };

  /**
   * Handles scrolling event of scroll element.
   * @param {Event} event Scroll event generated by user.
   */
  scrollHandler = event => {
    this.props.onScroll(event);
    this.updateHandler();
  };

  /**
   * Updates the scroll state of component.
   */
  updateScrollState = () =>
    this.setState({
      position: ScrollGradient.getScrollState(
        this.scrollContainer,
        this.props.direction
      ),
    });

  /**
   * @type {Function} Debounces the execution of scroll state update.
   */
  updateHandler = throttle(150, this.updateScrollState);

  /** @type {HTMLElement} Scrollable element reference. */
  scrollContainer = null;

  render() {
    const {
      className,
      children,
      direction,
      color,
      onScroll,
      scrollElementClassName,
      getScrollElementRef,
      hideStartGradient,
      ...other
    } = this.props;
    const { position } = this.state;
    const gradientRotation =
      direction === ScrollGradient.ScrollDirection.X ? -90 : 0;

    return (
      <div
        className={classnames(
          namespace,
          `${namespace}--${position.toLowerCase()}`,
          `${namespace}--${direction.toLowerCase()}`,
          className
        )}
        role="presentation"
        {...other}
      >
        {!hideStartGradient && (
          <div
            className={`${namespace}__before`}
            style={{
              backgroundImage: `linear-gradient(${gradientRotation}deg, rgba(0,0,0,0), ${color} 90%)`,
            }}
            role="presentation"
            aria-hidden
          />
        )}
        <div
          onScroll={this.scrollHandler}
          ref={this.setRefs}
          className={classnames(
            `${namespace}__content`,
            scrollElementClassName
          )}
        >
          {children}
        </div>
        <div
          className={`${namespace}__after`}
          style={{
            backgroundImage: `linear-gradient(${gradientRotation}deg, ${color} 10%, rgba(0,0,0,0))`,
          }}
          role="presentation"
          aria-hidden
        />
      </div>
    );
  }
}

export default ScrollGradient;
