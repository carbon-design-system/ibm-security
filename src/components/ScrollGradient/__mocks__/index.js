/**
 * @file `ResizeObserver` mock.
 * @copyright IBM Security 2020
 */

const { fn } = jest;

const disconnectMock = fn();
const observeMock = fn();

window.ResizeObserver = fn(() => ({
  disconnect: disconnectMock,
  observe: observeMock,
}));

export { disconnectMock, observeMock };
