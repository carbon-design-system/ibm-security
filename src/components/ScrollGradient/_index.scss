////
/// Scroll gradient component.
/// @group scroll-gradient
/// @copyright IBM Security 2019
////

@import '../Component/mixins';
@import 'mixins';

@include security--component($name: scroll-gradient) {
  @include scroll-gradient;
}
