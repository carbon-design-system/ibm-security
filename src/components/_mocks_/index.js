/**
 * @file Global mocks.
 * @copyright IBM Security 2018
 */

import React from 'react';

import random from '../../globals/random';

const label = 'Label';

const lorem =
  'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Eveniet rerum aliquid perferendis, nulla nam ad excepturi, iure earum dolorum ipsa quae cum voluptatibus nemo quis debitis, aperiam repudiandae dolore deleniti.';

const icon =
  'data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAzMiAzMiI+ICA8cGF0aCBmaWxsPSIjZmZmIiBkPSJNMTYgMThINmEyIDIgMCAwIDEtMi0yVjZhMiAyIDAgMCAxIDItMmgxMGEyIDIgMCAwIDEgMiAydjEwYTIgMiAwIDAgMS0yIDJ6TTYgNnYxMGgxMFY2em0yMCA2djRoLTR2LTRoNG0wLTJoLTRhMiAyIDAgMCAwLTIgMnY0YTIgMiAwIDAgMCAyIDJoNGEyIDIgMCAwIDAgMi0ydi00YTIgMiAwIDAgMC0yLTJ6bTAgMTJ2NGgtNHYtNGg0bTAtMmgtNGEyIDIgMCAwIDAtMiAydjRhMiAyIDAgMCAwIDIgMmg0YTIgMiAwIDAgMCAyLTJ2LTRhMiAyIDAgMCAwLTItMnptLTEwIDJ2NGgtNHYtNGg0bTAtMmgtNGEyIDIgMCAwIDAtMiAydjRhMiAyIDAgMCAwIDIgMmg0YTIgMiAwIDAgMCAyLTJ2LTRhMiAyIDAgMCAwLTItMnoiLz48L3N2Zz4=';

const id = '0';

const disabled = true;

const invalid = true;

const invalidText = 'Invalid text';

/**
 * Returns a child component.
 */
const children = <span>{label}</span>;

const className = 'class';

/**
 * Creates a URL for mock data.
 * @param {string} [hash=''] - A hash to pass to the URL method.
 * @returns {string} The mock data URL.
 */
const url = (hash = '') => `#${hash}`;

/**
 * Creates a `href` for mock data.
 * @returns {string} The `href` to return.
 */
const href = () => url();

export {
  children,
  className,
  disabled,
  href,
  icon,
  id,
  invalid,
  invalidText,
  label,
  lorem,
  random,
  url,
};
