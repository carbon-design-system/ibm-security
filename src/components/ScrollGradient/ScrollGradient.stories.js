/**
 * @file Scroll gradient stories.
 * @copyright IBM Security 2019
 */

import React from 'react';
import { storiesOf } from '@storybook/react';
import { color, boolean } from '@storybook/addon-knobs';
import { g100 } from '@carbon/themes';

import { components } from '../../../.storybook';
import { ScrollGradient } from '../..';
import { className, children } from './_mocks_';

const gradientColor = g100.uiBackground;

const style = {
  width: '300px',
  height: '300px',
};

storiesOf(components('ScrollGradient'), module)
  .add(
    'vertical scrolling',
    () => (
      <ScrollGradient
        color={color('color', gradientColor)}
        style={style}
        className={className}
      >
        {children}
      </ScrollGradient>
    ),
    {
      info: {
        text: `
        Vertical implementation of scroll gradient component that adds and removes top and bottom fade outs based on scrolling position.
      `,
      },
    }
  )
  .add(
    'horizontal scrolling',
    () => (
      <ScrollGradient
        color={color('color', gradientColor)}
        style={style}
        className={className}
        direction={ScrollGradient.ScrollDirection.X}
      >
        <div style={{ width: '200%' }}>{children}</div>
      </ScrollGradient>
    ),
    {
      info: {
        text: `
        Horizontal implementation of scroll gradient component that adds and removes left and right fade outs based on scrolling position.
      `,
      },
    }
  )
  .add(
    'with no start gradient',
    () => (
      <ScrollGradient
        color={color('color', gradientColor)}
        hideStartGradient={boolean('hideStartGradient', true)}
        style={style}
        className={className}
      >
        {children}
      </ScrollGradient>
    ),
    {
      info: {
        text: `
        Optionally hide start gradient on top or left depending on scroll direction.
      `,
      },
    }
  )
  .add(
    'with no scrolling content',
    () => (
      <ScrollGradient
        color={color('color', gradientColor)}
        style={style}
        className={className}
      >
        Content fits completely within content area, so neither scrollbar nor
        scroll gradients are necessary.
      </ScrollGradient>
    ),
    {
      info: {
        text: `
        This component does not render gradients if the content can fit completely within the allowed area.
      `,
      },
    }
  );
