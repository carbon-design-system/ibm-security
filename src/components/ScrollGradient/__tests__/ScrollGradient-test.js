/**
 * @file Scroll gradient tests.
 * @copyright IBM Security 2019 - 2021
 */

import React from 'react';
import { mount } from 'enzyme';

import ScrollGradient, { namespace } from '../ScrollGradient';

import { className, children } from '../_mocks_';

describe('ScrollGradient', () => {
  let scrollGradient;

  beforeEach(() => {
    scrollGradient = mount(
      <ScrollGradient
        className={className}
        color="blue"
        scrollElementClassName={className}
      >
        {children}
      </ScrollGradient>
    );
  });

  describe('Rendering', () => {
    it('renders gradient in y-direction by default', () =>
      expect(scrollGradient).toMatchSnapshot());

    it('renders gradient in x-direction', () => {
      scrollGradient.setProps({ direction: ScrollGradient.ScrollDirection.X });
      expect(scrollGradient).toMatchSnapshot();
    });

    it('passes scroll element reference via prop function', () => {
      const getScrollElementRef = jest.fn();

      // Refs are gotten when component is first mounted.
      scrollGradient = mount(
        <ScrollGradient
          className={className}
          color="blue"
          getScrollElementRef={getScrollElementRef}
        >
          {children}
        </ScrollGradient>
      );
      expect(getScrollElementRef).toHaveBeenCalledWith(expect.any(HTMLElement));
    });
  });

  describe('Events', () => {
    it('invokes `onScroll`', () => {
      const onScroll = jest.fn();
      scrollGradient.setProps({ onScroll });
      scrollGradient.find(`.${namespace}__content`).simulate('scroll');
      expect(onScroll).toHaveBeenCalledTimes(1);
    });

    describe('`ResizeObserver`', () => {
      const { fn } = jest;

      const disconnectMock = fn();
      const observeMock = fn();

      const { ResizeObserver } = window;

      beforeAll(() => {
        window.ResizeObserver = fn().mockImplementation(() => ({
          disconnect: disconnectMock,
          observe: observeMock,
        }));
      });

      afterAll(() => {
        window.ResizeObserver = ResizeObserver;
      });

      it('adds `ResizeObserver` when mounted', () => {
        expect(observeMock).toBeCalledTimes(1);
      });

      it('removes `ResizeObserver` when unmounted', () => {
        scrollGradient.unmount();

        expect(disconnectMock).toBeCalledTimes(1);
      });
    });
  });

  describe('getScrollState()', () => {
    it('returns `NONE` for non-scrollable x-direction', () =>
      expect(
        ScrollGradient.getScrollState(
          { clientWidth: 1, scrollWidth: 1 },
          ScrollGradient.ScrollDirection.X
        )
      ).toBe(ScrollGradient.ScrollStates.NONE));

    it('returns `INITIAL` when scrolling position is 0 for scrollable x-direction', () =>
      expect(
        ScrollGradient.getScrollState(
          { clientWidth: 1, scrollWidth: 3, scrollLeft: 0 },
          ScrollGradient.ScrollDirection.X
        )
      ).toBe(ScrollGradient.ScrollStates.INITIAL));

    it('returns `STARTED` when scrolling has started but has not reached end for scrollable x-direction', () =>
      expect(
        ScrollGradient.getScrollState(
          { clientWidth: 1, scrollWidth: 3, scrollLeft: 1 },
          ScrollGradient.ScrollDirection.X
        )
      ).toBe(ScrollGradient.ScrollStates.STARTED));

    it('returns `END` scrolled as far as possible in x-direction', () =>
      expect(
        ScrollGradient.getScrollState(
          { clientWidth: 1, scrollWidth: 3, scrollLeft: 2 },
          ScrollGradient.ScrollDirection.X
        )
      ).toBe(ScrollGradient.ScrollStates.END));

    it('returns `NONE` for non-scrollable y-direction', () =>
      expect(
        ScrollGradient.getScrollState(
          { clientHeight: 1, scrollHeight: 1 },
          ScrollGradient.ScrollDirection.Y
        )
      ).toBe(ScrollGradient.ScrollStates.NONE));

    it('returns `INITIAL` when scrolling position is 0 for scrollable y-direction', () =>
      expect(
        ScrollGradient.getScrollState(
          { clientHeight: 1, scrollHeight: 3, scrollTop: 0 },
          ScrollGradient.ScrollDirection.Y
        )
      ).toBe(ScrollGradient.ScrollStates.INITIAL));

    it('returns `STARTED` when scrolling has started but has not reached end for scrollable y-direction', () =>
      expect(
        ScrollGradient.getScrollState(
          { clientHeight: 1, scrollHeight: 3, scrollTop: 1 },
          ScrollGradient.ScrollDirection.Y
        )
      ).toBe(ScrollGradient.ScrollStates.STARTED));

    it('returns `END` scrolled as far as possible in y-direction', () =>
      expect(
        ScrollGradient.getScrollState(
          { clientHeight: 1, scrollHeight: 3, scrollTop: 2 },
          ScrollGradient.ScrollDirection.Y
        )
      ).toBe(ScrollGradient.ScrollStates.END));
  });
});
