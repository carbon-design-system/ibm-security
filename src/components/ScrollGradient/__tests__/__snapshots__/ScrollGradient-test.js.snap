// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollGradient Rendering renders gradient in x-direction 1`] = `
<ScrollGradient
  className="class"
  color="blue"
  direction="X"
  getScrollElementRef={[Function]}
  hideStartGradient={false}
  onScroll={[Function]}
  scrollElementClassName="class"
>
  <div
    className="security--scroll-gradient security--scroll-gradient--none security--scroll-gradient--x class"
    role="presentation"
  >
    <div
      aria-hidden={true}
      className="security--scroll-gradient__before"
      role="presentation"
      style={
        Object {
          "backgroundImage": "linear-gradient(-90deg, rgba(0,0,0,0), blue 90%)",
        }
      }
    />
    <div
      className="security--scroll-gradient__content class"
      onScroll={[Function]}
    >
      <p>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Eveniet rerum aliquid perferendis, nulla nam ad excepturi, iure earum dolorum ipsa quae cum voluptatibus nemo quis debitis, aperiam repudiandae dolore deleniti.
      </p>
      <p>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Eveniet rerum aliquid perferendis, nulla nam ad excepturi, iure earum dolorum ipsa quae cum voluptatibus nemo quis debitis, aperiam repudiandae dolore deleniti.
      </p>
      <p>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Eveniet rerum aliquid perferendis, nulla nam ad excepturi, iure earum dolorum ipsa quae cum voluptatibus nemo quis debitis, aperiam repudiandae dolore deleniti.
      </p>
    </div>
    <div
      aria-hidden={true}
      className="security--scroll-gradient__after"
      role="presentation"
      style={
        Object {
          "backgroundImage": "linear-gradient(-90deg, blue 10%, rgba(0,0,0,0))",
        }
      }
    />
  </div>
</ScrollGradient>
`;

exports[`ScrollGradient Rendering renders gradient in y-direction by default 1`] = `
<ScrollGradient
  className="class"
  color="blue"
  direction="Y"
  getScrollElementRef={[Function]}
  hideStartGradient={false}
  onScroll={[Function]}
  scrollElementClassName="class"
>
  <div
    className="security--scroll-gradient security--scroll-gradient--none security--scroll-gradient--y class"
    role="presentation"
  >
    <div
      aria-hidden={true}
      className="security--scroll-gradient__before"
      role="presentation"
      style={
        Object {
          "backgroundImage": "linear-gradient(0deg, rgba(0,0,0,0), blue 90%)",
        }
      }
    />
    <div
      className="security--scroll-gradient__content class"
      onScroll={[Function]}
    >
      <p>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Eveniet rerum aliquid perferendis, nulla nam ad excepturi, iure earum dolorum ipsa quae cum voluptatibus nemo quis debitis, aperiam repudiandae dolore deleniti.
      </p>
      <p>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Eveniet rerum aliquid perferendis, nulla nam ad excepturi, iure earum dolorum ipsa quae cum voluptatibus nemo quis debitis, aperiam repudiandae dolore deleniti.
      </p>
      <p>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Eveniet rerum aliquid perferendis, nulla nam ad excepturi, iure earum dolorum ipsa quae cum voluptatibus nemo quis debitis, aperiam repudiandae dolore deleniti.
      </p>
    </div>
    <div
      aria-hidden={true}
      className="security--scroll-gradient__after"
      role="presentation"
      style={
        Object {
          "backgroundImage": "linear-gradient(0deg, blue 10%, rgba(0,0,0,0))",
        }
      }
    />
  </div>
</ScrollGradient>
`;
